### ISSUE TITLE
Subtract minimum area from AUC metric

### ISSUE THREAD

**Issue by brethvoice** [tensorflow/tensorflow#50199]
**System information**
- TensorFlow version (you are using): nightly
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Current behavior is to report ROC and PR AUC as a raw number.  Since the theoretical AUC for ROC of a chance classifier is 0.5 and for PR is the ratio of true positives to all data points, make this an option for reporting.

**Will this change the current api? How?**
This will add a new optional Boolean argument, subtract_minimum_area, to the dunder init.

**Who will benefit with this feature?**
Everyone who wants to know whether their classifier is better than flipping a balanced coin.

**Any Other info.**
Code is on my fork at https://github.com/brethvoice/tensorflow

**Comment by UsharaniPagadala** [comment]
@brethvoice

 Could you please elaborate the issue with complete details,and what are the changes to be needed.Thanks

**Comment by brethvoice** [comment]
> @brethvoice
> 
> Could you please elaborate the issue with complete details,and what are the changes to be needed.Thanks

Complete details of changes requested (already in my fork):

1. Add new Boolean argument `subtract_minimum_area` to dunder init of AUC metric (default `False`)
2. If `False`, do nothing
3. If `True', subtract 0.5 from ROC AUC, or subtract proportion of true positives to overall number of data points from PR AUC

You can see the changes in the `result` method of the AUC class.  My [fork](https://github.com/brethvoice/tensorflow/blob/master/tensorflow/python/keras/metrics.py) only implements it for Python, and only in the case (for PR AUC) where:
`self.summation_method != metrics_utils.AUCSummationMethod.INTERPOLATION` so these details would have to be worked out.

**Comment by deeb02** [comment]
@brethvoice, for ROC curve this is simply subtracting 0.5, and anyone can do it for themselves. For PR AUC it may be somewhat useful, but I'm not convinced that it is used widely enough to warrant an API change.

**Comment by brethvoice** [comment]
> @brethvoice, for ROC curve this is simply subtracting 0.5, and anyone can do it for themselves. For PR AUC it may be somewhat useful, but I'm not convinced that it is used widely enough to warrant an API change.

@deeb02 it is definitely not used widely.  However, I wonder if that is simply because nobody has made it available?  I certainly would like to know when the PR-AUC is above its minimum value, and by how much.  Agreed about subtracting 0.5 from ROC AUC, however.
